src/main/java/teammates/ui/controller/StudentCommentsPageAction.java

```diff
@@ -153,7 +158,40 @@ private void setNextPageLink(List<CourseAttributes> courses, int currentIndex){
                 }
             }
         }
-        return feedbackResultBundles;
+        return sortFeedbackBundlesByEndTimeReverse(feedbackResultBundles);
+    }
+    
+    /**
+     * @param feedbackResultBundles
+     *            The map of FeedbackSessionResultsBundle which is to be sorted
+     * @return Returns a sorted map(LinkedHashMap) of
+     *         FeedbackSessionResultsBundle, where the sessions are sorted in
+     *         descending order of their endTime
+     */
+    private Map<String, FeedbackSessionResultsBundle> sortFeedbackBundlesByEndTimeReverse(
+            Map<String, FeedbackSessionResultsBundle> feedbackResultBundles) {
+
+        List<Entry<String, FeedbackSessionResultsBundle>> list = new ArrayList<Entry<String, FeedbackSessionResultsBundle>>(
+                feedbackResultBundles.entrySet());
+        Collections.sort(list,
+                new Comparator<Entry<String, FeedbackSessionResultsBundle>>() {
```

Comment:
Yeah I had seen this. But we will get a TreeSet< Entry< String, FeedbackSessionResultsBundle > >. Ultimately we need a  Map< String, FeedbackSessionResultsBundle >, which has a consistent iteration order(for use in the jsp file). So we will have to use a LinkedHashMap.